### DISCUSSION TITLE
Detect presence (and absence) of Tesla car through BLE iBeacons

### DISCUSSION THREAD

**Discussion by PedroKTFC** [tesla-local-control/tesla_ble_mqtt_docker#3] [Ideas]
Is it possible to detect the presence or otherwise of a Tesla car using BLE iBeacons? My use case is as follows:
- I have an electric gate I can open through Home Assistant
- When I get in my M3LR I'd like HA to detect someone is in the car and the gate should open.
- Similarly, when I get home the car should be detected as arrived and the gate open again.

**Comment by iainbullock** [comment]
I will have a look if it can be done. It might take me a while as I'm on holiday. If anyone else wants to comment on how best to do it or make a contribution to the code please do so (use dev branch for PRs)

**Reply by iainbullock** [reply]
Just to clarify I assume you want detect the presence of your car via its BLE rather than a (non Tesla) iBeacon device?

**Reply by PedroKTFC** [reply]
To be honest , whatever works best. Given it has BLE it would seem to be the most obvious, and presumably free, but if there's a better way I'd consider that.

**Comment by iainbullock** [comment]
Ok I'm working on something in the dev branch. It will provide a sensor which will indicate presence of the vehicle according to whether its BLE beacon is detectable

**Comment by iainbullock** [comment]
OK first working version is v1.0.7-dev. To test it make some changes to docker-compose.yml, and re-pull the image:

image: "iainbullock/tesla_ble_mqtt:latest" ->  image: "iainbullock/tesla_ble_mqtt:dev"

Add / change the line  - BLE_MAC=00:00:00:00:00:00 to reflect the BLE MAC you want to detect. A sesnor should appear in HA

Please can you test it? 

Thanks

**Reply by Briochorama** [reply]
Dumb question maybe, but how do you find the BLE MAC ?

**Comment by iainbullock** [comment]
Not dumb at all. I will document it if I implement into main. In the meantime run 

bluetoothctl scan on 

and look for it. 

From memory you will eventually see something like Device: XX:YY:ZZ:AA:BB:CC followed later on the same line by a string like Sa27xxxxxxxxa4e18C. 

Your BLE MAC is then XX:YY:ZZ:AA:BB:CC

CTRL-C to stop it
